bottlerocket/testsys/src/run_aws_ecs.rs

```diff
@@ -151,6 +152,15 @@ pub(crate) struct RunAwsEcs {
     /// provided, then the ECS test agent will attempt to create an IAM instance profile.
     #[structopt(long)]
     iam_instance_profile_name: Option<String>,
+
+    /// The path to a TOML file containing custom userdata.
+    #[structopt(long, requires("custom_user_data_mode"))]
+    custom_user_data: Option<String>,
+
+    /// The way custom userdata should interact with the default userdata.
+    /// The possible values are `merge` and `replace`.
+    #[structopt(long, requires("custom_user_data"))]
+    custom_user_data_mode: Option<String>,
 }
```

Comment:
We shouldn't add two optional fields to a struct that must both be `Some` in order for the structure to be in a consistent state. Instead we should have

```rust
struct CustomUserDataConfig {
    data: String,
    mode: CustomUserDataMode,
}


```

Then in the main struct:

```
    /// The path to a TOML file containing custom userdata.
    #[structopt(long, requires("custom_user_data_mode"))]
    custom_user_data: Option<CustomUserData>,
```

I know @ecpullen doesn't like this "tagging"-patterned use of an enum, but in this case since it has to work with yaml it might be the only way to make it self-consistent and self-documenting that both fields are required.

Reply:
Doesn't the `requires` `structopt` ensure that either none of them are present, or both of them are?

Reply:
That's how it's used in `cli` anyway.

Reply:
> Doesn't the requires structopt 

It might, but that doesn't prevent you from writing gross code when you go to use the fields and have to handle the inconsistent-state cases anyway (because they are Options and this is Rust)...

Reply:
`testsys` is going away, so I ok with any solution (even leaving it out).

Reply:
Sorry it took me a minute to realize this was an Args struct. You can ignore this comment thread.